test/e2e/far_e2e_test.go

```diff
@@ -41,26 +41,50 @@ const (
 
 var _ = Describe("FAR E2e", func() {
 	var (
-		far             *v1alpha1.FenceAgentsRemediation
-		fenceAgent      string
-		clusterPlatform *configv1.Infrastructure
-		err             error
+		fenceAgent, nodeIdentifierPrefix string
+		testShareParam                   map[v1alpha1.ParameterName]string
+		testNodeParam                    map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string
+		nodeIndex                        int
+		secondRun                        bool
 	)
 	BeforeEach(func() {
-		clusterPlatform, err = e2eUtils.GetClusterInfo(configClient)
+		// create FAR CR spec based on OCP platformn
+		clusterPlatform, err := e2eUtils.GetClusterInfo(configClient)
 		if err != nil {
 			Fail("can't identify the cluster platform")
 		}
 		fmt.Printf("\ncluster name: %s and PlatformType: %s \n", string(clusterPlatform.Name), string(clusterPlatform.Status.PlatformStatus.Type))
+
+		switch clusterPlatform.Status.PlatformStatus.Type {
+		case configv1.AWSPlatformType:
+			fenceAgent = fenceAgentAWS
+			nodeIdentifierPrefix = nodeIdentifierPrefixAWS
+			By("running fence_aws")
+		case configv1.BareMetalPlatformType:
+			fenceAgent = fenceAgentIPMI
+			nodeIdentifierPrefix = nodeIdentifierPrefixIPMI
+			By("running fence_ipmilan")
+		default:
+			Skip("FAR haven't been tested on this kind of cluster (non AWS or BareMetal)")
+		}
+
+		testShareParam, err = buildSharedParameters(clusterPlatform, fenceAgentAction)
+		if err != nil {
+			Fail("can't get shared information")
```

Comment:
Is there a reason for this `if err != nil {Fail(msg)}` pattern?
We usually use `Expect(err).ToNot(HaveOccured(), msg)`